src/dagmc/DagMC.cpp

```diff
@@ -212,6 +214,423 @@ ErrorCode DagMC::setup_indices() {
+  MB_CHK_SET_ERR(rval, "Failed to get the implicit complement");
+
+  EntityHandle inner_surface;
+  rval = box_to_surf(box.lower, box.upper, inner_surface);
+
+  // establish the volume-surface parent-child relationship with the inner
+  // surface
+  rval = MBI->add_parent_child(volume_set, inner_surface);
+  MB_CHK_SET_ERR(rval,
+                 "Failed to create the graveyard parent-child relationship");
+
+  // establish the volume-surface parent-child relationship with the inner
+  // surface
+  rval = MBI->add_parent_child(ic, inner_surface);
+  MB_CHK_SET_ERR(rval,
+                 "Failed to create the graveyard parent-child relationship");
+
+  // set the surface senses (all triangles have outward normals so this should
+  // be REVERSE wrt the graveyard volume)
+  EntityHandle inner_senses[2] = {ic, volume_set};
+  rval = MBI->tag_set_data(sense_tag(), &inner_surface, 1, inner_senses);
+  MB_CHK_SET_ERR(rval, "Failed to set graveyard surface senses");
+
+  // expand the box a bit again for the outer surface
+  for (int i = 0; i < 3; i++) {
+    double bump = 10.0 * numerical_precision();
+    box.upper[i] += bump;
+    box.lower[i] -= bump;
+  }
+
+  EntityHandle outer_surface;
+  rval = box_to_surf(box.lower, box.upper, outer_surface);
+
+  // establish the volume-surface parent-child relationship with the outer
+  // surface
+  rval = MBI->add_parent_child(volume_set, outer_surface);
+  MB_CHK_SET_ERR(rval,
+                 "Failed to create the graveyard parent-child relationship");
+
+  // establish the volume-surface parent-child relationship with the outer
+  // surface
+  rval = MBI->add_parent_child(ic, outer_surface);
+  MB_CHK_SET_ERR(rval,
+                 "Failed to create the graveyard parent-child relationship");
+
+  // set the surface senses (all triangles have outward normals so this should
+  // be FORWARD wrt the graveyard volume)
+  EntityHandle outer_senses[2] = {volume_set, ic};
+  rval = MBI->tag_set_data(sense_tag(), &outer_surface, 1, outer_senses);
+  MB_CHK_SET_ERR(rval, "Failed to set graveyard surface senses");
+
+  // OBBTree/BVH updates
+
+  // delete the implicit complement tree (but not the surface trees)
+  rval = geom_tool()->delete_obb_tree(ic, true);
+  MB_CHK_SET_ERR(rval, "Failed to delete the implicit complement tree");
+
+  // create BVH for both the new implicit complement and the new graveyard
+  // volume
+  rval = geom_tool()->construct_obb_tree(volume_set);
+  MB_CHK_SET_ERR(
+      rval,
+      "Failed to build accel. data structure for the new graveyard volume");
+
+  rval = geom_tool()->construct_obb_tree(ic);
+  MB_CHK_SET_ERR(
+      rval,
+      "Failed to build accel. data structure for the new implicit complement");
+
+  // re-initialize indices
+  rval = setup_indices();
+  MB_CHK_SET_ERR(rval, "Failed to setup indices after graveyard creation");
+
+  return rval;
+}
+
+ErrorCode DagMC::box_to_surf(double llc[3], double urc[3],
```

Comment:
`const &` for the `double[3]` arguments ?

Reply:
If it were a data structure that could get really large, then yes we'd want to do that. I don't think it's worth it here though.

Reply:
Making them const is probably good for peace of mind though :)